modules/imgproc/src/sumpixels.simd.hpp

```diff
@@ -180,61 +481,852 @@ struct Integral_SIMD<uchar, float, double>
 
-        // the others
-        for (int i = 0; i < height; ++i)
+        if (cn == 1)
         {
-            const uchar * src_row = src + _srcstep * i;
-            float * prev_sum_row = (float *)((uchar *)sum + _sumstep * i) + 1;
-            float * sum_row = (float *)((uchar *)sum + _sumstep * (i + 1)) + 1;
+            // the others
+            for (int i = 0; i < height; ++i)
+            {
+                const uchar * src_row = src + _srcstep * i;
+                double * prev_sum_row = (double *)((uchar *)sum + _sumstep * i) + 1;
+                double * sum_row = (double *)((uchar *)sum + _sumstep * (i + 1)) + 1;
 
-            sum_row[-1] = 0;
+                sum_row[-1] = 0;
 
-            v_float32 prev = vx_setzero_f32();
-            int j = 0;
-            for (; j + v_uint16::nlanes <= width; j += v_uint16::nlanes)
+                v_float64 prev = vx_setzero_f64();
+                int j = 0;
+                for (; j + v_uint16::nlanes <= width; j += v_uint16::nlanes)
+                {
+                    v_int16 el8 = v_reinterpret_as_s16(vx_load_expand(src_row + j));
+                    v_float64 el4ll, el4lh, el4hl, el4hh;
+#if CV_AVX2 && CV_SIMD_WIDTH == 32
+                    __m256i vsum = _mm256_add_epi16(el8.val, _mm256_slli_si256(el8.val, 2));
+                    vsum = _mm256_add_epi16(vsum, _mm256_slli_si256(vsum, 4));
+                    vsum = _mm256_add_epi16(vsum, _mm256_slli_si256(vsum, 8));
+                    __m256i el4l_32 = _mm256_cvtepi16_epi32(_v256_extract_low(vsum));
+                    __m256i el4h_32 = _mm256_cvtepi16_epi32(_v256_extract_high(vsum));
+                    el4ll.val = _mm256_add_pd(_mm256_cvtepi32_pd(_v256_extract_low(el4l_32)), prev.val);
+                    el4lh.val = _mm256_add_pd(_mm256_cvtepi32_pd(_v256_extract_high(el4l_32)), prev.val);
+                    __m256d el4d = _mm256_permute4x64_pd(el4lh.val, 0xff);
+                    el4hl.val = _mm256_add_pd(_mm256_cvtepi32_pd(_v256_extract_low(el4h_32)), el4d);
+                    el4hh.val = _mm256_add_pd(_mm256_cvtepi32_pd(_v256_extract_high(el4h_32)), el4d);
+                    prev.val = _mm256_permute4x64_pd(el4hh.val, 0xff);
+#else
+                    el8 += v_rotate_left<1>(el8);
+                    el8 += v_rotate_left<2>(el8);
+#if CV_SIMD_WIDTH >= 32
+                    el8 += v_rotate_left<4>(el8);
+#if CV_SIMD_WIDTH == 64
+                    el8 += v_rotate_left<8>(el8);
+#endif
+#endif
+                    v_int32 el4li, el4hi;
+                    v_expand(el8, el4li, el4hi);
+                    el4ll = v_cvt_f64(el4li) + prev;
+                    el4lh = v_cvt_f64_high(el4li) + prev;
+                    el4hl = v_cvt_f64(el4hi) + el4ll;
+                    el4hh = v_cvt_f64_high(el4hi) + el4lh;
+                    prev = vx_setall_f64(v_extract_n<v_float64::nlanes - 1>(el4hh));
+//                    prev = v_broadcast_element<v_float64::nlanes - 1>(el4hh);
```

Comment:
Why removed `v_broadcast_element()`?

Reply:
v_broadcast_element for v_float64 is not available for all platforms.